src/api-calls/game-log/removeLog.ts

```diff
@@ -0,0 +1,29 @@
+import { createApiFunction } from "api-calls/createApiFunction";
+import { deleteDoc } from "firebase/firestore";
+import {
+  getCampaignGameLogDocument,
+  getCharacterGameLogDocument,
+} from "./_getRef";
+
+export const removeLog = createApiFunction<
+  { campaignId?: string; characterId?: string; logId: string },
+  void
+>((params) => {
+  const { campaignId, characterId, logId } = params;
+
+  return new Promise((resolve, reject) => {
+    if (!characterId && !campaignId) {
+      reject(new Error("Either campaign or character ID must be defined."));
+    }
+
+    const docRef = characterId
+      ? getCharacterGameLogDocument(characterId, logId)
+      : getCampaignGameLogDocument(campaignId as string, logId);

```

Comment:
I think swapping this ternary to focus on campaignIds will fix the issue of GMs not being able to delete logs.

```
campaignId ? getCampaignGameLogDocument() : getCharacterGameLogDocument();
```

Right now, if a character ID is set, it will always try and delete from the character, even if the campaign is defined, which is why its not working for GMs on the character sheet in a campaign.